src/app/features/auth/store/auth.effects.ts

```diff
@@ -0,0 +1,60 @@
+import { inject } from '@angular/core';
+import { Router } from '@angular/router';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { CookieService } from 'ngx-cookie-service';
+import { ToastrService } from 'ngx-toastr';
+import { catchError, filter, map, of, startWith, switchMap, tap } from 'rxjs';
+
+import { APP_PATH } from 'src/app/app.module';
+import { HOME_PATHS } from '../../home';
+import { AuthService } from '../auth.service';
+import { AuthActions } from './auth.actions';
+
+export class AuthEffects {
+  private actions$ = inject(Actions);
+  private cookieService = inject(CookieService);
+  private toast = inject(ToastrService);
+  private authService = inject(AuthService);
+  private router = inject(Router);
+
+  login$ = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(AuthActions.login),
+      switchMap(({ loginData }) => {
+        const { email, password } = loginData;
+        return this.authService.login(email, password).pipe(
```

Comment:
operatory z tego pipe mozna wyniesc pietro wyzej za `switchMap`, jedyne co trzeba zostawic to jak sie nie myle `catchError`, ale to trzeba byłoby sprawdzić